src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -134,6 +134,39 @@ public override bool Equals(object obj)
     /// </summary>
     public abstract partial class WorkspaceModel : NotificationObject, ILocatable, IUndoRedoRecorderClient, ILogSource, IDisposable, IWorkspaceModel
     {
+        #region nested classes
+
+        /// <summary>
+        /// This class enables the delay of graph execution.
+        /// Use instances of this class to specify a code scope in which you want graph execution to be delayed. 
+        /// Class is thread safe, although behavior is not well defined. 
+        /// Nested instance of this class do not have a well defined behavior.
+        /// </summary>
+        internal class DelayedGraphExecution : IDisposable
```

Comment:
hmm - I like the new design, what about DelayedWorkspaceExecution? Or is that an implementation detail rather than an API one?

Reply:
hmm, I'm not convinced that is better or worse